inc/classes/abstracts/class-imagify-abstract-attachment.php

```diff
@@ -33,7 +33,21 @@ class Imagify_Abstract_Attachment {
 	 * @param  int $id The attachment ID.
 	 * @return void
 	 */
-	public function __construct( $id = 0 ) {}
+	public function __construct( $id = 0 ) {
+		global $post;
+
+		if ( $id ) {
+			if ( is_a( $id, 'WP_Post' ) ) {
+				$this->id = $id->ID;
+			} elseif ( is_numeric( $id ) ) {
+				$this->id = $id;
+			}
+		} elseif ( $post && is_a( $post, 'WP_Post' ) ) {
+			$this->id = $post->ID;
+		}
+
+		$this->id = absint( $this->id );
+	}
```

Comment:
Should an abstract class have any code in it? Isn't it supposed to lay ground for the base structure only?